### ISSUE TITLE
Error when WHERE clause only contains BIND

### ISSUE THREAD

**Issue by jakubklimek** [openlink/virtuoso-opensource#294]
Commit: 9deeba78df3ae2273cb63e66c7469f93edaa18e7

<pre><code>select ?dec where {
BIND( strdt("50.08332824707031",xsd:double) as ?dec)
}</code></pre>

gives:

<pre><code>Virtuoso 37000 Error SP031: SPARQL compiler: The list of return values contains '*' but the pattern does not contain variables

SPARQL query:
define sql:big-data-const 0 
#output-format:text/html
define sql:signal-void-variables 1 select ?dec where {

BIND( strdt("50.08332824707031",xsd:double) as ?dec)
}</code></pre>

but there is no \* and there are variables

**Comment by kidehen** [comment]
We need to see if this is a VOS issue based on the following examples 
that don't have the problem:

http://linkeddata.uriburner.com/sparql?default-graph-uri=&query=select+%3Fdec+where+%7B%0D%0ABIND%28+strdt%28%2250.08332824707031%22%2Cxsd%3Adouble%29+as+%3Fdec%29%0D%0A%7D&should-sponge=&format=text%2Fhtml&CXML_redir_for_subjs=121&CXML_redir_for_hrefs=&timeout=30000000

Don't get it on DBpedia:
http://dbpedia.org/sparql?default-graph-uri=&query=select+%3Fdec+where+%7B%0D%0ABIND%28+strdt%28%2250.08332824707031%22%2Cxsd%3Adouble%29+as+%3Fdec%29%0D%0A%7D&should-sponge=&format=text%2Fhtml&CXML_redir_for_subjs=121&CXML_redir_for_hrefs=&timeout=30000000

**Comment by HughWilliams** [comment]
@jakubklimek: looking at the commit, is this error occurring when querying via the Jena provider as that is what it is for? But then I thought  you guys use Sesame?

As when I test against the `develop/7` build I made last night, it does not have this problem:
```
SQL> SPARQL select ?dec where { BIND( strdt("50.08332824707031",xsd:double) as ?dec) };
dec
VARCHAR

---

50.08332824707031

1 Rows. -- 5 msec.
SQL> status('');
REPORT
VARCHAR
```
---
```
OpenLink Virtuoso  Server
Version 07.10.3211-pthreads for Linux as of Jan 14 2015 
Started on: 2015-01-14 20:33 GMT+1
```

**Comment by jakubklimek** [comment]
I tried this one via the sparql endpoint

**Comment by HughWilliams** [comment]
@jakubklimek: OK, seems to be related to the **Strict checking of void variables** checkbox on the SPARQL endpoint, as if you uncheck it the query works. 

Can also recreated from `isql` by simulating the setting of this check:
```
SQL> SPARQL define sql:signal-void-variables 1   select ?dec where {  BIND( strdt("50.08332824707031",xsd:double) as ?dec) };

**\* Error 37000: VD [Virtuoso Server]SQ074: Line 1: SP031: SPARQL compiler: The list of return values contains '*' but the pattern does not contain variables
at line 56 of Top-Level:
SPARQL define sql:signal-void-variables 1   select ?dec where {  BIND( strdt("50.08332824707031",xsd:double) as ?dec) }
SQL> 
```
Have reported to development to look into ...

**Comment by jakubklimek** [comment]
Probably same as https://github.com/openlink/virtuoso-opensource/issues/231

**Comment by drassokhin** [comment]
I ran into the same problem with `define sql:signal-void-variables 1` and `BIND` in the query a couple of days ago. Also got `The list of return values contains '*' but the pattern does not contain variables` error. It almost drove me nuts as I spent almost an hour trying to understand what was wrong with my pretty complicated SPARQL query before I finally figured out the `sql:signal-void-variables 1` option triggered it. I was using `Version 07.20.3217-threads for Win64 as of Apr 25 2016`.

**Comment by HughWilliams** [comment]
@drassokhin: Thus issue is resolved in the upcoming Virtuoso 8 commercial release ... development seeing how it can be resolved in the current Virtuoso 7.2 release ...

**Comment by jakubklimek** [comment]
@hughwilliams Is there going to be open source version 8 as well?

**Comment by HughWilliams** [comment]
@jakubklimek: No there will a  open source Virtuoso 7 maintenance shortly, we will seek to make this fix available in if possible ...

**Comment by jakubklimek** [comment]
Fixed in d76ab20f8cdee8b7660bf1b653fda85d08e5153c
